### DISCUSSION TITLE
Using Claude Sonet 3.5

### DISCUSSION THREAD

**Discussion by solarslurpi** [community/community#143173] [Copilot Conversations] [answered]
### Select Topic Area

Using other models than Open AI's lineup.

### Body

I thought we could use other models like Claude 3.5. I find this model the best for chatting up the code.  When will this be supported? I don't find the same success to GPT 4, etc.

**Comment by joemar25** [comment]
Try this guide bro. [Guide](https://zapier.com/blog/claude-api/).

You can see the API key by following documentation in that blob, and use the model in your app.

![image](https://github.com/user-attachments/assets/5f8d4dba-9328-4d8a-bf1c-8d4589c40495)

![image](https://github.com/user-attachments/assets/98f03c8e-e213-4645-8539-779d2ad4b71e)

But you might put your credit card to use it.

![image](https://github.com/user-attachments/assets/bbd2b7a6-a6f6-4600-9268-b6d8561d2101)

**Reply by solarslurpi** [reply]
yes. i use the model in the app via the api. I am doing that. If that is what you mean. What I cannot seem to do is set up copilot to use claude3.5 sonnet. 
![image](https://github.com/user-attachments/assets/19ba6753-3dfb-4a42-a134-91abe302dcf2)

**Reply by Panky55** [reply, marked as answer]
have you updated vscode and the copilot extension? also have you tried switching to the pre release version of the extension? if that doesn't work then i guess its still rolling out to people since ive already got it

**Reply by joemar25** [reply]
![image](https://github.com/user-attachments/assets/15f405e0-b0f5-4e58-8b9d-d6c8b1ff3c11)

**Reply by joemar25** [reply]
Here is documentation for it. [docs](https://docs.github.com/en/copilot/using-github-copilot/using-claude-sonnet-in-github-copilot).
Yeah, kindly update and then do this base on the documentation 'You must enable access to Claude 3.5 Sonnet before you can use the model.'

![image](https://github.com/user-attachments/assets/4528b4f0-1983-4f7d-bd58-3ff3b50cb173)

**Reply by solarslurpi** [reply]
Thank you. Indeed I am one of the "have nots"
